### ISSUE TITLE
Incorrect Gradient for a normalization function

### ISSUE THREAD

**Issue by avik-pal** [FluxML/Zygote.jl#197]
```julia
julia> using Zygote, ForwardDiff

julia> a = rand(10);

julia> normalize(x) = sum((x .- minimum(x)) ./ (maximum(x) .- minimum(x)))

julia> normalize'(a)
10-element Array{Float64,1}:
  1.1689102183698927
  1.1689102183698927
  1.1689102183698927
  1.1689102183698927
  1.1689102183698927
  1.1689102183698927
  1.1689102183698927
 -3.6509926281379084
  1.1689102183698927
  1.1689102183698927

julia> ForwardDiff.gradient(normalize, a)
10-element Array{Float64,1}:
  1.1689102183698927
  1.1689102183698927
 -5.700289118821235 
  1.1689102183698927
  1.1689102183698927
  1.1689102183698927
  1.1689102183698927
 -3.6509926281379066
  1.1689102183698927
  1.1689102183698927
```

It is not difficult to define a custom adjoint but it might be worth looking into this weird gradient at the index where the value is maximum.

**Comment by MikeInnes** [comment]
Any luck debugging this? Would be great to have a patch.

**Comment by avik-pal** [comment]
I have a [custom adjoint](https://github.com/avik-pal/RayTracer.jl/blob/master/src/gradients/zygote.jl#L186-L199) to fix this but not sure if that is the correct way forward.

**Comment by MikeInnes** [comment]
Ideally we should just fix the adjoint for `maximum` itself. The Flux version seems to work, can you narrow down the difference between Zygote's version and that?

**Comment by avik-pal** [comment]
The gradient for maximum seems to be fine to me. However, the gradient for `gradient(x -> 1 / maximum(x), arr)` is `(nothing,)`. Which is not expected!

**Comment by MikeInnes** [comment]
Closed with #208
